tests/integration-tests/conftest.py

```diff
@@ -1394,12 +1397,23 @@ def _run_benchmarks(remote_command_executor, scheduler_commands, **kwargs):
                         test_datadir,
                         dimensions,
                     )
+                    # Load reference results to confront benchmark and Assert failure
+                    if check_result:
+                        reference_results = results_csv_to_dict()
+                        check_errors = []
+
                     for metric_data in metric_data_list:
+                        # Check threshold
+                        if check_result:
+                            check_errors = check_threshold(reference_results, metric_data)
```

Comment:
I'd move this logic in the `if key == "check_result":` to avoid an extra by rewriting the code with:

```python
if key == "check_result" and value:
     reference_results = results_csv_to_dict()
```

and moving check_errors initialization outside of all the loops, in this way you'll have a check_errors empty list to return in any case and make the function consistent (always returning the same info).